kube/deployment.go

```diff
@@ -99,20 +98,41 @@ func generalCheck(role *model.Role, controller *helm.Mapping, settings ExportSet
 	// also means that we now have to guard ourselves against use
 	// of the old keys. Here we add the necessary guard
 	// conditions.
+	//
+	// Note. The construction shown for requests and limits is used because of
+	// (1) When FOO is nil you cannot check for FOO.BAR, for any BAR.
+	// (2) The `and` operator is not short cuircuited, it evals all of its arguments
+	// Thus `and FOO FOO.BAR` will not work either
 
 	for _, key := range []string{
 		"HA",
-		"cpu.limits",
-		"cpu.requests",
-		"memory.limits",
-		"memory.requests",
+		"cpu",
+		"memory",
 	} {
-		guardVariable := fmt.Sprintf("_moved_sizing_%s", strings.Replace(key, ".", "_", -1))
-		block := fmt.Sprintf("if .Values.sizing.%s", key)
-		fail := fmt.Sprintf(`{{ fail "Bad use of moved variable sizing.%s. The new name to use is config.%s" }}`,
-			key, key)
+		if key == "HA" {
+			guardVariable := fmt.Sprintf("_moved_sizing_%s", key)
+			block := fmt.Sprintf("if .Values.sizing.%s", key)
+			fail := fmt.Sprintf(`{{ fail "Bad use of moved variable sizing.%s. The new name to use is config.%s" }}`,
+				key, key)
+
+			controller.Add(guardVariable, fail, helm.Block(block))
+			continue
+		}
```

Comment:
Not sure why you put this block inside the loop, and then just filter it out. Just move it on top of the loop, and get rid of the conditional. Also saves the `fmt.Sprintf` to setup `fail` because the `key` is already known.

There will never be any additional entries for this loop, as this is a one-time migration, and there are no other non-rolename keys left in `.Values.sizing`, so no reason to not make this code as simple as possible.

Reply:
```
			fail := `{{ fail "Bad use of moved variable sizing.HA. The new name to use is config.HA" }}`
			controller.Add("_moved_sizing_HA", fail, helm.Block("if .Values.sizing.HA"))
```

Reply:
Somehow my mind was wedded to the loop. Agreed, specialized this one case is more sensible.
To come.

Reply:
Done with commit 6a2949581a367e0a8e743750e1dca34f32446451